src/store/theme.js

```diff
@@ -0,0 +1,57 @@
+import React, { Component } from 'react';
+
+export const ThemeContext = React.createContext();
+
+export function withThemeContext(Element) {
+    return function WrapperComponent(props) {
+        return (
+            <ThemeContext.Consumer>
+                {state => (
+                    <Element
+                        {...props}
+                        context={{ ...state, ...props.context }}
+                    />
+                )}
+            </ThemeContext.Consumer>
+        );
+    };
+}
+
+class ThemeProvider extends Component {
+    constructor(props) {
+        super(props);
+
+        this.toggleTheme = () => {
+            this.setState(state => ({
+                theme: !state.theme
+            }));
+            const { theme } = this.state;
+            localStorage.setItem('theme', theme);
+        };
+
+        this.getTheme = () => {
+            const { theme } = this.state;
+            return theme
+                ? { bg: '#1b1919', fg: 'white' }
+                : { bg: 'white', fg: '#1b1919' };
+        };
+
+        this.state = {
+            theme: !!localStorage.getItem('theme'),
+            toggleTheme: this.toggleTheme,
```

Comment:
Maybe handleChangeTheme or just changeTheme no ?

Reply:
why not, but i think toggleTheme is good as well tho

Reply:
toogle theme for me means I'm gonna turn on and off something, in this case turn on the theme and turn off, thats way the suggestion

Reply:
yes you rigth, you going to turn on or off the dark theme